🆕: Add new endpoint on account module.

MarcusSilvaDev opened the pull request.

MarcusSilvaDev committed:
🆕: Add new endpoint on account module.

caioavidal commented on src/Standalone/appsettings.Local.json:
```diff
@@ -9,6 +9,6 @@
       "SQLITE": "Data Source=neo.db",
       "POSTGRESQL": "host=localhost;port=5432;database=opencoremmo;username=postgres;password=postgres;"
     },
-    "active": "INMEMORY"
```
can you turn it back to in memory?

caioavidal commented on src/Standalone/appsettings.json:
```diff
@@ -44,7 +44,7 @@
       "SQLITE": "Data Source=neo.db",
       "POSTGRESQL": "host=localhost;port=5432;database=opencoremmo;username=postgres;password=postgres;"
     },
-    "active": "INMEMORY"
```
same

MarcusSilvaDev committed:
🆕: Add new columns account table and new table to accountpremium history.

MarcusSilvaDev committed:
💉: create integration tests for account module.

MarcusSilvaDev commented:
### PR Description  
This PR introduces new features and improvements to the account module, including new endpoints, database updates, and integration tests to ensure code quality.  

### Changes Implemented:  
1. **New Endpoints in the Account Module**  
   - Added endpoints to:  
     - Create an account.  
     - Ban an account.  
     - Change account password.  
     - Add premium to an account.  

2. **Database Structure Updates**  
   - Added new columns to the `account` table.  
   - Created a new table to store the account premium history.  

3. **Integration Tests**  
   - Developed integration tests for the account module to validate the newly implemented features.  

### Tests Performed:  
- Successfully executed integration tests for all new endpoints and database updates.  

### Checklist  
- [x] Endpoints tested and functional: Create Account, Ban Account, Change Password, Add Premium.  
- [x] Database structure changes validated.  
- [x] Integration tests added and passing.  


Evidences:


![image](https://github.com/user-attachments/assets/1d30c31c-4a50-4b74-8d55-89d234db5ac9)

![image](https://github.com/user-attachments/assets/64e31499-c144-4b21-a10e-443d8c9724f3)

caioavidal commented on src/Database/NeoServer.Data/Configurations/AccountEntityConfiguration.cs:
```diff
@@ -62,7 +69,8 @@ private static void Seed(EntityTypeBuilder<AccountEntity> builder)
                 PremiumTime = 30,
-                AllowManyOnline = true
+                AllowManyOnline = true,
+                AccountName = "GOD"
```
can you change accountName to also "1" please?
